### PR TITLE
[bugfix/frontend] Preload css styles

### PR DESCRIPTION
**Pull request by tsmethurst**
# Description

> If this is a code change, please include a summary of what you've coded, and link to the issue(s) it closes/implements.
>
> If this is a documentation change, please briefly describe what you've changed and why.

This pull request implements [preloading](https://devdocs.io/html/link_types/preload) for css in our header template. Should hopefully reduce instances of [FOUC](https://en.wikipedia.org/wiki/Flash_of_unstyled_content).

Screenshot from 2023-03-20 18-32-15

## Checklist

Please put an x inside each checkbox to indicate that you've read and followed it: `[ ]` -> `[x]`

If this is a documentation change, only the first checkbox must be filled (you can delete the others if you want).

- [x] I/we have read the [GoToSocial contribution guidelines](https://github.com/superseriousbusiness/gotosocial/blob/main/CONTRIBUTING.md).
- [x] I/we have discussed the proposed changes already, either in an issue on the repository, or in the Matrix chat.
- [x] I/we have performed a self-review of added code.
- [x] I/we have written code that is legible and maintainable by others.

### CODE REVIEW

File: web/template/header.tmpl [hunk 1]
```diff
@@ -42,7 +42,10 @@
 	<meta property="og:image:height" content="{{ .ogMeta.ImageHeight }}">
 	{{ end }}{{ end }}<link rel="shortcut icon" href="{{ .instance.Thumbnail }}" type="{{ if .instance.ThumbnailType }}{{ .instance.ThumbnailType }}{{ else }}image/png{{ end }}">
 	{{ if .rssFeed }}<link rel="alternate" type="application/rss+xml" href="{{ .rssFeed }}" title="{{ if .ogMeta }}{{ .ogMeta.Title }}{{ else }}{{.instance.Title}}{{ end }}">{{ end }}
-	<link rel="stylesheet" href="/assets/dist/_colors.css">
+	<link rel="preload" href="/assets/dist/_colors.css" as="style">
+	<link rel="preload" href="/assets/dist/base.css" as="style">
+	{{range .stylesheets}}<link rel="preload" href="{{.}}" as="style">
+	{{end}}<link rel="stylesheet" href="/assets/dist/_colors.css">
```
[the remark this answers is not in the document]
Comment (tsmethurst) [reply web/template/header.tmpl:48]: Sort of! One of them defines the preload and one of them defines the stylesheet. It wasn't clear to me whether or not the preloads had to be before the stylesheets, so to be on the safe side i ranged through twice.

[the remark this answers is not in the document]
Comment (daenney) [reply web/template/header.tmpl:48]: Ah, right you are: https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel/preload#the_basics

[the remark this answers is not in the document]
Comment (daenney) [reply web/template/header.tmpl:48]: It seems there is a world in which `rel="stylesheet preload"` ought to work, but we don't live in that timeline.

[the remark this answers is not in the document]
Comment (tsmethurst) [reply web/template/header.tmpl:48]: alas :( you think this is OK to merge now?

[the remark this answers is not in the document]
Comment (daenney) [reply web/template/header.tmpl:48]: I'm not very versed in frontend-y things, but the change makes sense given the MDN docs. So I'd say :shipit: and worst case we revert it.

**Review by daenney — APPROVED** [review]

**Comment by EchedeyLR** [comment]
Was this tested against WebKitGTK-based web browsers such as GNOME Web?

Seems the feature is not supported so I wonder if this will cause an issue while rendering on them.

**Comment by tsmethurst** [comment]
> Was this tested against WebKitGTK-based web browsers such as GNOME Web?
> 
> Seems the feature is not supported so I wonder if this will cause an issue while rendering on them.

It wasn't, no. However! It's currently deployed on goblin.technology, so if you can open that site in such a browser, then we're all good :)

**Comment by EchedeyLR** [comment]
I tested in both GNOME Web and Pale Moon, both seemed okay with the instance.

I scrolled in your profile.

However, I noticed that the scrolling was quite rough in GNOME Web over GtS, something I only experimented in Pleroma before.

The hardware of this puter is quite good but I guess this is a different issue and unrelated to prefetch/preload.

**Comment by f0x52** [comment]
GNOME Web (scrolling) performance has always felt quite terrible to me regardless of site content, and static templates like profiles really aren't doing much that'd cost performance afaik

**Comment by EchedeyLR** [comment]
Coming from some junior dev experience yet, I can say this is quite untrue and is more a fault of some web devs.

I have worked on several websites with some web frameworks (Angular and React) and my versions were performant on WebKitGTK.

It is true that I usually test side-to-side with that web browser where most people just test bleeding edge Chrome and Firefox "stable".

**Comment by daenney** [comment]
I would suggest filing a separate issue for WebkitGTK perf. This PR was about preloading assets. It seems to have achieved that and not breaking anything. The additional preload directive that's being ignored by WebkitGTK isn't likely to be the source of any performance issue.

**Comment by EchedeyLR** [comment]
What you are writing in your comment @daenney is something I already wrote in my message previous to the @f0x52 reply...

**Pull request merged**